### PR TITLE
slides: mount editor on mobile + Pointer Events migration

### PR DESCRIPTION
**Pull request by hackerwins**
## Summary

Phase B Task 1a + 1b of [Slides Mobile Light Edit](../blob/slides/mobile-edit-pointer-events/docs/design/slides/slides-mobile-edit.md). Splits cleanly into two cohesive halves:

**Task 1a — Pointer Events migration (slides package only)**
- Mechanical rename of `mouse*` event listeners to `pointer*` across the slides editor module (`editor.ts`, `thumbnail-panel.ts`, `context-menu.ts`, `layout-picker.ts`). 5 test files updated to dispatch `PointerEvent` instead of `MouseEvent`.
- `PointerEvent` is a strict superset of `MouseEvent` (it inherits from it in TS); browsers synthesize pointer events from both mouse and touch input. Desktop behavior is unchanged.
- Prerequisite for touch drag on iOS Safari, which never synthesizes `mousemove` during a touch drag (only `mousedown`/`mouseup` for taps). Pen tablets and stylus input get supported as a side-effect.

**Task 1b — Mount full SlidesEditor on mobile**
- `MobileSlidesView` gains a `mode: 'view' | 'edit'` prop. `'view'` keeps Phase A's read-only `SlideRenderer` with swipe nav. `'edit'` (default) builds canvas + overlay and mounts the desktop `SlidesEditor`.
- New `touchHandleTolerance?: number` option on `SlidesEditor`. `handleHitTest` now expands hit rectangles by that many pixels on each side AND switches to closest-center disambiguation when expanded rects overlap (fingers on small selections). Desktop default is 0; mobile passes 22 (≈ 44px Apple HIG min).
- Two iOS-specific CSS guards on canvas-host: `touch-action: none` (canvas owns horizontal pointer — no browser swipe/pinch competition) and `-webkit-touch-callout: none` + `user-select: none` (kills long-press preview that `contextmenu` blocking doesn't cover).
- Mutations flow through the existing `SlidesStore`, so Yorkie sync and undo/redo are inherited from desktop with zero new mutation surface.

Permission-gated `'view'` fallback for shared-link viewers is Task 4 (separate PR). Bottom-sheet text formatting + slide-ops FAB are Tasks 2-3.

## Test plan

- [x] `pnpm verify:fast` green (792 slides tests + 2 new hit-test tolerance/overlap tests).
- [x] Desktop regression smoke (puppeteer 1400×900): select via dispatched `PointerEvent`, drag a shape 100px, verify nw handle moved and screenshot confirms element followed.
- [x] iPhone 16 Pro simulator (Safari): user-driven matrix — tap-select ✅, drag-move ✅, drag-resize ✅, drag-rotate ✅, long-press callout suppressed ✅, double-tap text-edit ✅, blank-tap clear ✅. Korean IME ❌ — deferred to Task 2 (text-edit-focused PR) since it needs docs-side IME work.
- [x] Real device (physical iPhone) pass for Korean IME and pinch behavior — deferred to Task 2.
- [x] Android Chrome touch pass — deferred to Task 1b real-device follow-up.

## Self-review fixes already applied

Pre-push review (via superpowers:requesting-code-review) caught and fixed:
- **Critical:** mobile edit-mode cleanup now calls `editor.detach()` to match `slides-view.tsx:479` — was leaking document-level pointer/key listeners per mount cycle.
- **Important:** `onCurrentSlideChange` callback uses functional `setState` to avoid stale-closure compare against `currentSlideId`.

Deferred with rationale (logged in [lessons](docs/tasks/active/20260517-slides-mobile-edit-lessons.md)):
- Global `[data-handle]` style selector is a pre-existing desktop pattern (`slides-view.tsx:301-303`); follow-up scopes both.
- jsdom `PointerEvent.pointerId` defaults to 0 vs real-browser 1+ — only matters if a future PR adds `setPointerCapture`.

## Design + planning

- Design doc: `docs/design/slides/slides-mobile-edit.md`
- Todo: `docs/tasks/active/20260517-slides-mobile-edit-todo.md` (Tasks 0/1a/1b done; 2-4 follow as separate PRs)
- Lessons: `docs/tasks/active/20260517-slides-mobile-edit-lessons.md`

## Summary by CodeRabbit

* **New Features**
  * Added mobile editor mode to replace read-only viewing with interactive editing on touch devices.

* **Improvements**
  * Migrated to Pointer Events for improved touch interaction and drag reliability on mobile.
  * Enhanced handle hit-testing with tolerance adjustment for better touch accuracy.

* **Documentation**
  * Added design and implementation documentation for mobile editing features.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wafflebase/wafflebase/pull/261?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: docs/design/slides/slides-mobile-edit.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+```
+                     before (Phase A)
+  YorkieSlidesStore ──► (used only by Present)
+       SlideRenderer ──► <canvas>   (read-only paint)
+
+                     after (Phase B)
+  YorkieSlidesStore ──► SlidesEditor ──► <canvas> + <overlay>
+                                    └──► getActiveTextEditor() ──► <BottomSheet>
+                                    └──► store.{addSlide, undo, ...}
+```
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply docs/design/slides/slides-mobile-edit.md]: Fixed in 61a9c4dc — both fences tagged `text`.

File: docs/tasks/active/20260517-slides-mobile-edit-todo.md [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+- [ ] **1a.4** Run `pnpm verify:fast`. Expect zero new failures —
+  the existing slides editor unit tests use synthetic events; if
+  any test dispatches a literal `new MouseEvent('mousedown', ...)`
+  the test still works because pointer-event listeners do NOT fire
+  on dispatched mouse events. **Re-check the test files** for any
+  such cases and update them to dispatch `PointerEvent` if so. Use:
+
+  ```bash
+  grep -rnE "new MouseEvent\(|dispatchEvent.*mouse" packages/slides/src --include="*.ts" --include="*.test.ts"
+  ```
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply docs/tasks/active/20260517-slides-mobile-edit-todo.md:147]: Fixed in 61a9c4dc — grep widened to `packages/slides` so test files are included. The migration actually did need to touch test files (5 of them), so the original scope would have hidden that work.
